core-backend/wasmtime/src/env.rs

```diff
@@ -299,22 +332,16 @@ impl<E: Ext + IntoExtInfo> Environment<E> for WasmtimeEnvironment<E> {
 
         let res = entry_func.call(&mut self.store, &[], &mut []);
 
+        let termination_reason = self.store.data().termination_reason.clone();
```

Comment:
Found bug - `TerminationReason::GasAllowanceExceeded` from `Ext::charge_gas*` is not considered

Reply:
Considered actually